src/Impostor.Api.Innersloth.Generator/Generators/EnumGenerator.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Text.Json;
+using CSharpPoet;
+using Microsoft.CodeAnalysis;
+
+namespace Impostor.Api.Innersloth.Generator.Generators;
+
+public sealed class EnumGenerator : BaseGenerator
+{
+    public EnumGenerator(SourceProductionContext sourceProductionContext, ImmutableArray<(string RelativePath, string Content)> files) : base(sourceProductionContext, files)
+    {
+    }
+
+    public void Generate(string name, string? @namespace = null, string? sourceName = null, bool flags = false, CSharpEnumUnderlyingType underlyingType = CSharpEnumUnderlyingType.Int)
+    {
+        var dictionary = JsonSerializer.Deserialize<Dictionary<string, long>>(GetFileContent($"enums/{sourceName ?? name}.json"))!;
+
+        var @enum = new CSharpEnum(name, underlyingType);
+
+        foreach (var pair in dictionary)
+        {
+            var value = flags && pair.Value > 0
+                ? $"1 << {Math.Log(pair.Value, 2)}"
```

Comment:
Why do you explicitly want to represent flags as 1 << x? It silences errors when a value is not a power of 2

Reply:
Why not?

> It silences errors when a value is not a power of 2

The build will fail anyway, and we can handle it better when it becomes a problem.
![image](https://github.com/Impostor/Impostor/assets/35262707/493c4405-91a5-4b34-ae9b-382040f02b1d)

Reply:
Ah, I was afraid it'd floor the value, which would cause issues